docs/developer/pull-requests.md

```diff
@@ -3,6 +3,142 @@ This document outlines best practices to follow when creating and reviewing PRs.
 
 ## As an author
 
+### Pull request process
+1. Create a GitHub PR from your fork
+1. Verify all checks are passing
+1. Get someone to review it
+1. Iterate your changes based on the feedback you receive. Don't rewrite the history (rebase or squash) with your new changes, keep them as individual commits. This facilitates the reviewers job.
+1. Once you haver al least one `lgtm` and are ready to merge, squash your commits into one.
```

Comment:
```suggestion
1. Once you have al least one `lgtm` and are ready to merge, squash your commits into one.
```

Reply:
auto squash on merge, no?

Reply:
I think we can avoid the squash completely and just use auto squash if we have a good rule on the commit the makes the PR subject because I think the PR subject gets used for the squash, no? I'd have to try it out to be sure.

Reply:
Any other options to configure how auto squash works. Nobody has time to squash commits as the final step, I just don't think that is reasonable and while that commit history is not great I just don't think it is worth the extra manual effort to squash.

Reply:
I, strongly, disagree. I squash manually all the time. `git rebase -i HEAD~<commits to squash>` followed by some vim magic `%s/pick/f/gc` using `gc` options so I can confirm things. 

It literally takes <30s.

Reply:
That said, this workflow doesn't work with the existing setup because the minute you squash and force push `lgtm` is removed and you're left to bug people for lgtm again. Thats a bad flow.

Reply:
If losing the `lgtm` label is the only concern, we could use [StoreTreeHash](https://pkg.go.dev/k8s.io/test-infra/prow/plugins#Lgtm) in the prow config. That should prevent the `lgtm` label from being removed if we are only squashing

Reply:
Lets do it and see where it goes.